telco-examples/edge-clusters/dhcp/edge-metallb-multi-node/telco-capi-metallb-multi-node.yaml

```diff
@@ -0,0 +1,222 @@
+    pods:
+      cidrBlocks:
+        - 192.168.0.0/18
+    services:
+      cidrBlocks:
+        - 10.96.0.0/12
+  controlPlaneRef:
+    apiVersion: controlplane.cluster.x-k8s.io/v1alpha1
+    kind: RKE2ControlPlane
+    name: multinode-cluster
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+    kind: Metal3Cluster
+    name: multinode-cluster
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: Metal3Cluster
+metadata:
+  name: multinode-cluster
+  namespace: default
+spec:
+  controlPlaneEndpoint:
+    host: ${EDGE_VIP_ADDRESS}
+    port: 6443
+  noCloudProvider: true
+---
+apiVersion: controlplane.cluster.x-k8s.io/v1alpha1
+kind: RKE2ControlPlane
+metadata:
+  name: multinode-cluster
+  namespace: default
+spec:
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+    kind: Metal3MachineTemplate
+    name: multinode-cluster-controlplane
+  replicas: 3
+  registrationMethod: "address"
+  registrationAddress: ${EDGE_VIP_ADDRESS}
+  serverConfig:
+    cni: cilium
+    tlsSan:
+      - ${EDGE_VIP_ADDRESS}
+      - https://${EDGE_VIP_ADDRESS}.sslip.io
+  agentConfig:
+    format: ignition
+    additionalUserData:
+      config: |
+        variant: fcos
+        version: 1.4.0
+        systemd:
+          units:
+            - name: rke2-preinstall.service
+              enabled: true
+              contents: |
+                [Unit]
+                Description=rke2-preinstall
+                Wants=network-online.target
+                Before=rke2-install.service
+                ConditionPathExists=!/run/cluster-api/bootstrap-success.complete
+                [Service]
+                Type=oneshot
+                User=root
+                ExecStartPre=/bin/sh -c "mount -L config-2 /mnt"
+                ExecStart=/bin/sh -c "sed -i \"s/BAREMETALHOST_UUID/$(jq -r .uuid /mnt/openstack/latest/meta_data.json)/\" /etc/rancher/rke2/config.yaml"
+                ExecStart=/bin/sh -c "echo \"node-name: $(jq -r .name /mnt/openstack/latest/meta_data.json)\" >> /etc/rancher/rke2/config.yaml"
+                ExecStartPost=/bin/sh -c "umount /mnt"
+                [Install]
+                WantedBy=multi-user.target
+            - name: metallb-endpointcopier-operators.service
+              enabled: true
+              contents: |
+                [Unit]
+                Description=metallb-endpointcopier-operators
+                Wants=network-online.target
+                After=rke2-server.service
+                [Service]
+                Type=oneshot
+                User=root
+                ExecStartPre=/bin/sh -c "while [ ! -d /var/lib/rancher/rke2/server/manifests ]; do sleep 1; done"
+                ExecStart=/bin/sh -c "cp /var/metallb-endpointcopier-operators.yaml /var/lib/rancher/rke2/server/manifests/metallb-endpointcopier-operators.yaml"
+                [Install]
+                WantedBy=multi-user.target
+            - name: metallb-cr.service
```

Comment:
I don't think we need to do this on every control plane host. I'm not sure of the consequences of this being on 3 different places.

Also, I don't know if instead we should deploy both metallb and endpointcopieroperator as HelmCharts instead of extra manifests. Doing so we will make sure there is only one (`kubectl get helmchart -n kube-system metallb || kubectl apply...`) and it will be easier to manage for updates/upgrades/modifications (just edit the helmchart object instead of modifying the _file_)
/cc @hardys

Reply:
Not sure if I understand the second part, but I'm doing with helmcharts (lines 115-132)

Reply:
Yes, you are doing it with helmcharts as extra manifests https://docs.rke2.io/advanced#auto-deploying-manifests My _concern_ with this approach is that when you want to make a change to the charts in order to either upgrade or change a value, you need to modify the file itself, not the K8s object.
Also, adding those files at RKE2ControlPlane level means those files will be present on all the control-plane hosts... which can have side effects I guess (I'm just speculating, I didn't tried but it doesn't seem a good idea to me).

What I was suggesting was to have a systemd unit that does a kubectl apply of the helmchart object instead... and only on one of the hosts

Reply:
@Kristian-ZH is going to test this part in order to know what's happening (to see if 3 controllers are configured or only 1) I don't have enough servers to test it, so after talking with Kristian, he is going to test this part. 
let's wait until he can test it

Reply:
See the DANGER section here https://docs.k3s.io/installation/packaged-components:
```
If you have multiple server nodes, and place additional AddOn manifests on more than one server, it is your responsibility to ensure that files stay in sync across those nodes. K3s does not sync AddOn content between nodes, and cannot guarantee correct behavior if different servers attempt to deploy conflicting manifests.
```

Reply:
as we talk, let's keep as is by now, to see which is the best strategy and i will change later on in case we need it